cmd/vela-server/validate.go

```diff
@@ -140,6 +140,16 @@ func validateQueue(c *cli.Context) error {
 func validateSecret(c *cli.Context) error {
 	logrus.Trace("Validating secret CLI configuration")
 
+	// nolint // ignoring line length
+	if len(c.String("native-key")) == 0 {
+		// enforce AES-256, so check explicitly for 32 bytes on the key
+		if len(c.String("native-key")) != 32 {
+			return fmt.Errorf("native-key invalid length: %d", len(c.String("native-key")))
+		}
+
+		return fmt.Errorf("native-key (VELA_SECRET_NATIVE_KEY or SECRET_NATIVE_KEY) flag not specified")
+	}
```

Comment:
I wonder if we have to change this a little?

If I'm reading this correctly, [on line 144](https://github.com/go-vela/server/blob/secrets/crypto/cmd/vela-server/validate.go#L144), we check if the length of the value for the `native-key` flag **is equal to** `0`.

But then, [on line 146](https://github.com/go-vela/server/blob/secrets/crypto/cmd/vela-server/validate.go#L146), we check if the length of the value for the `native-key` flag **is not equal to** `32`.

I might be missing it, but the way I see this play out in my head is we don't go into the `if` block until the value for the `native-key` flag **is equal to** `0`.. but that means it will **always not equal** `32`.

I believe this means if a user is to spin up the server and doesn't have the `native-key` flag set, they'll never see the `flag not specified` error.

They'll always be met with the `native-key invalid length` error message.

This also means that if a value is provided for the `native-key` flag, then we'll never check to see if it's equal to `32` bytes like I believe we want.

Here is an example of how we might rewrite this:

```suggestion
	// nolint // ignoring line length
	if len(c.String("native-key")) == 0 {
		return fmt.Errorf("native-key (VELA_SECRET_NATIVE_KEY or SECRET_NATIVE_KEY) flag not specified")
	}
	
	// enforce AES-256, so check explicitly for 32 bytes on the key
	if len(c.String("native-key")) != 32 {
		return fmt.Errorf("native-key invalid length: %d", len(c.String("native-key")))
	}
```

This code will first check to see if the length of the value for the `native-key` flag **is equal to** `0`.

If it is, then we immediately return an error up the stack.

If it's not, then we continue with the next validation check which checks to see if the length of the value for the `native-key` flag **is not equal to** `32`.

Thoughts?

Reply:
Simplifed